cmd/root.go

```diff
@@ -41,5 +41,5 @@ func init() {
 	cmd.Flags().StringVar(&cfg.DockerContext, "context", "", "Context is the path to the build context")
 	cmd.Flags().StringVar(&cfg.Destination, "destination", "", "Destination is the destination of the built image")
 	cmd.Flags().StringVar(&cfg.RegistryMirrors, "registry-mirrors", "", "Registry mirrors to find images")
-	cmd.Flags().BoolVar(&cfg.SkipDefaultRegistryFallback, "skip-default-registry-fallback", false, "Fail if image is not found on registry mirrors")
+	cmd.Flags().StringVar(&cfg.SkipDefaultRegistryFallback, "skip-default-registry-fallback", "false", "Fail if image is not found on registry mirrors")
```

Comment:
Instead of turning the boolean flag into a string flag, I suggest to specify a default value for the Action input [here](https://github.com/cloudbees-io/kaniko/blob/60551b9b511b1f4f1f52c5745577b02eea75ecba/action.yml#L30):
```
  skip-default-registry-fallback:
    default: 'false'
```

Reply:
From what I tested, using `skip-default-registry-fallback: 'false'` will still set the flag. I think cobra is interpreting a bool flag as true by existing as an arg, not what the value is.

I will run a test to see what happens if the args are instead set as: `--skip-default-registry-fallback="${{ inputs.skip-default-registry-fallback }}"`

Reply:
> I think cobra is interpreting a bool flag as true by existing as an arg, not what the value is.

I don't think that's how cobra behaves. It respects the value of the CLI option. The only special behaviour that the boolean cobra flag comes with is that you can omit the value of the option which makes cobra treat it as `true`.

Reply:
When I run:
```
with:  
  dockerfile: Dockerfile-test
  destination: registry.saas-dev.beescloud.com/staging/kaniko-action-test:${{ cloudbees.scm.sha }}
  skip-default-registry-fallback: 'false'
```
And command line: `args: ... --skip-default-registry-fallback "${{ inputs.skip-default-registry-fallback }}"`

The flag is still present when executing the kaniko binary:
```
Running command: /kaniko/executor --verbosity=debug --ignore-path=/cloudbees/ --dockerfile Dockerfile-test 
--context /cloudbees/workspace 
--destination registry.saas-dev.beescloud.com/staging/kaniko-action-test:b6553c53bdaa7016f49a51d0e33338baa62ab9de 
--digest-file /tmp/kaniko-image-digest --skip-default-registry-fallback
```
I verified that `--skip-default-registry-fallback="${{ inputs.skip-default-registry-fallback }}"` does pass boolean false and the flag is not present on the command to kaniko.

Reply:
That doesn't make sense to me. How did you test that? Is it possible that you used a different image version that is not aligned (same git commit) with the changes you made to the action.yml?

I just tested your binary changes to verify/show that it behaves as expected:
```sh
$ go build -o kaniko-action .
$ mkdir bin
$ ln -s /usr/bin/echo bin/executor
$ export PATH="`pwd`/bin:$PATH"
$ ./kaniko-action --skip-default-registry-fallback=false
2024/02/22 19:38:29 found kaniko executor binary at /home/max/development/kaniko-action/bin/executor
Running command: /home/max/development/kaniko-action/bin/executor --verbosity=debug --ignore-path=/cloudbees/ --destination 
--verbosity=debug --ignore-path=/cloudbees/ --destination
```

As you can see the providing the option `--skip-default-registry-fallback=false` sets the flag to `false` as expected.
In fact I was using that new feature/input earlier this week successfully using [this branch](https://github.com/cloudbees-io/kaniko/tree/sdp-14029) where I forcefully aligned the action.yml with the image version [like this](https://github.com/cloudbees-io/kaniko/blob/064b62e00e5a371a9460fa865ece88e9eff10967/action.yml#L66).

Reply:
I realize you used the `action.scm.sha` expression within the destination image reference of your test action invocation mentioned in your previous comment as opposed to using that within the `uses` field. That may be the problem.